### PR TITLE
fix: improve error functionality for asset data-point and event observability mode

### PR DESCRIPTION
**Pull request by vilit1**
This does the following:
1. fix the `az iot ops asset update` custom attribute param
2. make the cli check if the asset data point or event has a valid observability mode and set the default to "none" (without having to wait for the service to do so)
4. "fix" up the integration tests to include cluster/custom location resource groups

Too tired to think of a meme so here is a deer:
image

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

File: cluster_query.json
Comment (vilit1) [inline cluster_query.json:1]: this is what happens when you run tests as you commit

**Comment by vilit1** [comment]
image

File: azext_edge/edge/providers/rpsaas/adr/user_strings.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 # Asset Strings
 ENDPOINT_NOT_FOUND_WARNING = "Endpoint {0} not found. The asset may fail provisioning."
+INVALID_OBSERVABILITY_MODE_ERROR = "{0} has an invalid observability mode [{1}]."
```
Comment (c-ryan-k) [inline azext_edge/edge/providers/rpsaas/adr/user_strings.py:9]: Should this error tell them which observability mode is invalid? It seems to be used to show the valid modes in the array `[{1}]`, but it doesn't show which one is wrong. Maybe something like : `{0} has invalid observability mode "{1}", valid modes are: [{2}]` is more descriptive?

Comment (vilit1) [reply azext_edge/edge/providers/rpsaas/adr/user_strings.py:9]: there is one observability mode per event or data point. So I used the event notifier/data source (aka what is required) to help the customer identify which one is invalid. 

I didn't think there is any need to echo the customer input for the wrong observability mode back but if you feel strongly about this, I can add it in.

**Review by c-ryan-k — APPROVED** [review]

**Review by c-ryan-k — APPROVED** [review]

**Pull request merged**